sources/common/dbdump.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+)
+
+type BaseDbDump interface {
+	GetBaseDdl() BaseToDdl
+	ProcessDump(conv *internal.Conv, r *internal.Reader) error
+}
```

Comment:
Consider dropping Base from the interface names since these are not really base classes but rather meaningful interfaces. Thus, it would change following:

Also, it's typical to name concrete implementations with a Impl suffix e.g., mysql.DbDumpImpl

BaseDbDump -> DbDump
with concrete implementations : mysql.DbDumpImpl, postgres.DbDumpImpl 

(Notice how it avoids stuttering in the previous naming i.e., mysql.MysqlDbDump -- mysql is repeated twice)

Similarly for other interfaces

Reply:
Done